zigzag/opt/loma/LomaEngine.py

```diff
@@ -211,12 +215,68 @@ def compute_nb_permutations(self):
             f"{self.nb_permutations:,}",
         )
 
+    def reduce_static_fps(self):
+        static_sizes: list[tuple[LayerDim, int]] = []
+        for constr in self.constraints:
```

Comment:
What is the purpose of this loop? Do you want any `static_sizes` of `self.constraints`, or specifically the last one?

Reply:
I only want static_sizes from the StaticPositionsAndSizesConstraint class, but because the constraints list has constraints of all types belonging to the superclass, we need to loop through the list to filter only get the correct constraint. Only a single constraint of each type is allowed in each list, so there will only be one possible value for static_sizes. If there is no StaticPositionsAndSizesConstraint in constraints, an empty list is returned